### PR TITLE
Restructure XRC_01 example for contract demo 

### PR DESCRIPTION
**Pull request by qizheng09**
## Description

What is the purpose of the change?

This commit Restructure the example of XRC_01.

For detailed information, please #421

### CODE REVIEW

File: core/contractsdk/cpp/XRC/xrc01/src/main/xrc01_exemple.cc [hunk 1]
```diff
@@ -0,0 +1,414 @@
+#include <assert.h>
+#include "xchain/xchain.h"
+#include "xrc01.h"
+
+// XRC是超级链协议标准族
+// XRC_01是XRC协议家族中的第一个协议，支持在超级链中发行通过资产
```
Comment (yucaowang) [inline core/contractsdk/cpp/XRC/xrc01/src/main/xrc01_exemple.cc]: "发行通过资产" typo?

File: core/contractsdk/cpp/XRC/xrc01/src/main/xrc01_exemple.cc [hunk 2]
```diff
@@ -0,0 +1,414 @@
...
+};
+
+struct XRC01_Demo : public XRC01_Example, public xchain::Contract {
+public:
+    bool str2bool(const std::string var, bool& var_bool) {
+        if (var == "1") {
+            var_bool = false;
+            return true;
+        } else if (var == "0") {
+            var_bool = true;
```
Comment (yucaowang) [inline core/contractsdk/cpp/XRC/xrc01/src/main/xrc01_exemple.cc]: common sense is "1" == true and "0" == false, why is it reversed here?

File: core/contractsdk/cpp/XRC/xrc01/src/main/xrc01_exemple.cc [hunk 3]
```diff
@@ -0,0 +1,414 @@
...
+            return true;
+        }
+        return false;
+    }
+
+    bool safe_stoull(const std::string in, uint64_t* out) {
+        if (in == "") {
```
Comment (yucaowang) [inline core/contractsdk/cpp/XRC/xrc01/src/main/xrc01_exemple.cc]: suggest use string.empty() to check if the param is empty string

File: core/contractsdk/cpp/XRC/xrc01/src/main/xrc01_exemple.cc (hunk 1 above)
Comment (qizheng09) [reply core/contractsdk/cpp/XRC/xrc01/src/main/xrc01_exemple.cc]: thks, fixed

File: core/contractsdk/cpp/XRC/xrc01/src/main/xrc01_exemple.cc (hunk 2 above)
Comment (qizheng09) [reply core/contractsdk/cpp/XRC/xrc01/src/main/xrc01_exemple.cc]: Thanks, fixed

File: core/contractsdk/cpp/XRC/xrc01/src/main/xrc01_exemple.cc (hunk 3 above)
Comment (qizheng09) [reply core/contractsdk/cpp/XRC/xrc01/src/main/xrc01_exemple.cc]: Thanks, fixed

**Pull request merged**